### PR TITLE
Add players, create rolls, and stats board

### PR DESCRIPTION
**Pull request by jdmcleod**
## Why?

Begin creating front end.

## What Changed

Please list the changes in this PR?

* Created add player functionality
* Created add roll functionality
* Created stats board
Screen Shot 2019-08-28 at 1 04 35 PM

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (Jeremy-Walton) [inline .vscode/settings.json]: You should add the `.vscode` folder to a `.gitignore` file so your personal settings don't get committed.

File: app.rb [hunk 2]
```diff
@@ -30,7 +31,26 @@ class MyApp < Sinatra::Base
...
+    @calculator.find_or_add_player(name)
+    slim :index
+  end
+
+  post '/roll' do
+    @num1 = params[:player_one_num_dice]
+    @num2 = params[:player_two_num_dice]
+    @winner = params[:winner]
+    @loser = params[:loser]
+    @rawInput = @winner + " " + @loser + " " + @num1 + " " + @num2
```
Comment (Jeremy-Walton) [inline app.rb]:
```suggestion
    @rawInput = "#{@winner} #{@loser} #{@num1} #{@num2}"
```

File: app.rb [hunk 3]
```diff
@@ -30,7 +31,26 @@ class MyApp < Sinatra::Base
...
+    puts @rawInput
```
Comment (Jeremy-Walton) [inline app.rb]: Do we need to print to the console anymore since we are making a front end?

File: app.rb [hunk 4]
```diff
@@ -30,7 +31,26 @@ class MyApp < Sinatra::Base
...
+    @calculator = settings.calculator
+    @calculator.runScenario(@rawInput)
     slim :index
```
Comment (Jeremy-Walton) [inline app.rb]: Should this be redirecting back to the index with the results?

File: views/index.slim [hunk 5]
```diff
@@ -1,5 +1,72 @@
-h1 Welcome!
-
-ul
-  - @calculator.players.each do |player|
-    li #{player.name}
+doctype html
```
Comment (Jeremy-Walton) [inline views/index.slim]: Any reason you are using the layout here?

File: app.rb [hunk 6]
```diff
@@ -1,4 +1,5 @@
 require 'sinatra/base'
+require('sinatra/reloader')
```
Comment (Jeremy-Walton) [inline app.rb]: What is this for? Where does it come from

Comment (Jeremy-Walton) [reply app.rb]: Looks like it's not included in the base package so in order to use this (if we actually need it) we would need to add `sinatra-contrib` to the gemfile. See http://sinatrarb.com/contrib/reloader

File: views/index.slim [hunk 7]
```diff
@@ -1,5 +1,72 @@
...
+html
+  head
+    title Risk Calculator
+    link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.1/css/bootstrap.min.css"
+    link rel='stylesheet' type='text/css' href='css/application.css'
+  body
+    script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"
+    script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"
+    javascript:
+        $("button").click(function(){
```
Comment (Jeremy-Walton) [inline views/index.slim]: What is this needed for?

File: views/index.slim [hunk 8]
```diff
@@ -1,5 +1,72 @@
...
+          $("button").removeClass("active");
+          $(this).addClass("active");
+        });   
+    .container.rolls
+      .row
+        .col-md-6
+          .panel.panel-info#panel1
+            form action="/new_player" method="post" class="add-player"
+              .input-group.spacer
+                input id="n" name="new_name" class="form-control" type="text" placeholder="name"
```
Comment (Jeremy-Walton) [inline views/index.slim]: Since this input doesn't have a label, it doesn't need an id

File: views/index.slim [hunk 9]
```diff
@@ -1,5 +1,72 @@
...
+                .col-md-3
+                  button type="submit" class="btn btn-success run-scenerio" Proccess Roll
+        .col-md-6
+          .panel.panel-info#panel2
+        .col-md-12
+          .table-wrapper
+            h1.stats Stats
+            table.fl-table
+              tr
+                th colspan="4" Player
```
Comment (Jeremy-Walton) [inline views/index.slim]: You can use single quotes here and all the following

File: views/layout.slim [hunk 10]
```diff
@@ -1,7 +1,7 @@
 doctype html
 html
   head
-    title Risk Calculator
+    title Risk Calculator Thing
```
Comment (Jeremy-Walton) [inline views/layout.slim]: Why add Thing? As per my comment above, why aren't we using the layout for the index page?

File: views/stylesheets/application.scss [hunk 11]
```diff
@@ -3,10 +3,122 @@
 $black: black;
 $white: white;
 
+.run-scenerio {
```
Comment (Jeremy-Walton) [inline views/stylesheets/application.scss]: There are much better ways to style this but it will be easier to pair with you and redo it

File: views/stylesheets/bootstrap.css [hunk 12]
```diff
@@ -0,0 +1,6757 @@
+/*!
```
Comment (Jeremy-Walton) [inline views/stylesheets/bootstrap.css]: It's good that you added bootstrap directly, but we are loading it from a cdn, see the layout file or the head in index.slim. Those should be removed since we have this now and this should be loaded in it's place.

File: app.rb [hunk 13]
```diff
@@ -30,7 +31,26 @@ class MyApp < Sinatra::Base
 
   get '/' do
     @calculator = settings.calculator
-    @calculator.runScenario('Jeremy Josh 3 1') # This populates the players so you can actually see something
+    @calculator.runScenario('Jeremy Josh 3 1')
```
Comment (Jeremy-Walton) [inline app.rb]: Since you have the ability to add players now, do we need this initial scenario anymore?

File: app.rb [hunk 14]
```diff
@@ -30,7 +31,26 @@ class MyApp < Sinatra::Base
...
+    slim :index
+  end
+
+  post '/new_player' do
+    name = params[:new_name]
+    @calculator = settings.calculator
+    @calculator.find_or_add_player(name)
```
Comment (Jeremy-Walton) [inline app.rb]: You should update `find_or_add_player` to reject or ignore empty names. Currently, if you just hit add player without anything in the field, it will add a player with no name. You can also add input validation on the front end as well

File: app.rb [hunk 15]
```diff
@@ -30,7 +31,26 @@ class MyApp < Sinatra::Base
...
+    slim :index
+  end
+
+  post '/roll' do
+    @num1 = params[:player_one_num_dice]
```
Comment (Jeremy-Walton) [inline app.rb]: Don't abrev. Abbreviations cause confusion and are usually a sign of bad variable names.

`roll_one`, `roll_two` would be just fine in this situation instead of ``player_one_num_dice

File: views/index.slim [hunk 16]
```diff
@@ -1,5 +1,72 @@
...
+        });   
+    .container.rolls
+      .row
+        .col-md-6
+          .panel.panel-info#panel1
+            form action="/new_player" method="post" class="add-player"
+              .input-group.spacer
+                input id="n" name="new_name" class="form-control" type="text" placeholder="name"
+                span class="input-group-btn"
+                  button type="submit" class="btn btn-info add-player" Add Player
```
Comment (Jeremy-Walton) [inline views/index.slim]: There is a lot of cleanup that can happen here as well, but again it will be easier to do if we pair.

**Review by Jeremy-Walton — CHANGES_REQUESTED** [review]
Lots of things to consider

**Review by Jeremy-Walton — APPROVED** [review]
